### PR TITLE
Create new FHIR profiles

### PR DESCRIPTION
**Pull request by simonvadee**
client:
- add a new button "create profile" when adding a resource
- this button uploads a structuredefinition to fhir-api (if valid)
- validation is applied on the profile (are the types compatible, validate json schema and required fields...)
- refresh the cache of pyrog since the newly uploaded definition does not exist in redis yet

server:
- add a new resolver "refreshDefinition" to fetch a definition from fhir-api and cahce it in redis

### CODE REVIEW

File: client/src/components/uploadProfile/index.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      const profile = JSON.parse(reader.result as string);
+      if (validateProfile(profile)) {
+        setProfileDefinition(profile);
+        setinvalidProfileErrors(undefined);
+      }
+    };
+    reader.readAsText(profileFile);
+  }, [acceptedFiles]);
+
+  const renderError = () => {
```
Comment (Jasopaum) [inline client/src/components/uploadProfile/index.tsx]: I don't think it's needed to have a function here

File: client/src/components/uploadProfile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        <ul>
+          {invalidProfileErrors!.map((err, i) => (
+            <li key={i}>{err} </li>
+          ))}
+        </ul>
+      </div>
+    );
+  };
+
+  const renderProfileOverview = () => {
```
Comment (Jasopaum) [inline client/src/components/uploadProfile/index.tsx]: Same here

File: server/src/fhir/definitions.ts [hunk 3]
```diff
@@ -72,20 +74,6 @@ export const bootstrapDefinitions = async () => {
   )
   await Promise.all(standardDefinitions.items.map(cacheDefinition))
   console.log('Done.')
-
```
[the remark this answers is not in the document]
Comment (simonvadee) [reply server/src/fhir/definitions.ts]: I do but it upsets me to see the comments here, I just created an issue to find a clever way to bootstrap definitions

File: client/src/components/uploadProfile/index.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    rejectedFiles
+  } = useDropzone({
+    multiple: false,
+    accept: 'application/json'
+  });
+  const [profileDefinition, setProfileDefinition] = React.useState(
+    undefined as any
+  );
+  const [invalidProfileErrors, setinvalidProfileErrors] = React.useState(
+    undefined as string[] | undefined
```
[the remark this answers is not in the document]
Comment (simonvadee) [reply client/src/components/uploadProfile/index.tsx]: ok done

File: client/src/components/uploadProfile/index.tsx (hunk 1 above)
Comment (simonvadee) [reply client/src/components/uploadProfile/index.tsx]: done.

File: client/src/components/uploadProfile/index.tsx (hunk 2 above)
Comment (simonvadee) [reply client/src/components/uploadProfile/index.tsx]: i do need a function here

File: client/src/components/selects/addResourceSelect.tsx [hunk 5]
```diff
@@ -55,6 +96,32 @@ const AddResourceSelect = ({
     );
   };
 
+  const renderProfileList = (
+    resource: Resource
+  ): ItemListRenderer<Profile> => ({
+    items,
```
Comment (Jasopaum) [inline client/src/components/selects/addResourceSelect.tsx]: nit: you don't use items nor query

**Pull request merged**